sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/configuration/CommonConfiguration.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.aad.configuration;
+
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.converter.FormHttpMessageConverter;
+import org.springframework.security.oauth2.core.http.converter.OAuth2AccessTokenResponseHttpMessageConverter;
+import org.springframework.web.client.RestOperations;
+
+import java.util.Arrays;
+
+/**
+ * Common bean used for Azure AD and Azure AD B2C
+ */
+@Configuration(proxyBeanMethods = false)
+public class CommonConfiguration {
```

Comment:
This name is too general, and can this class be package-private?

Reply:
If it can't be package-private, can we move it to the impl package?

Reply:
I'd suggest something like `AadRestOperationConfiguration` to narrow the scope of this class.

Reply:
> I'd suggest something like AadRestOperationConfiguration to narrow the scope of this class.

Will rename to `AadRestOperationConfiguration `.

Reply:
> This name is too general, and can this class be package-private?
> If it can't be package-private, can we move it to the impl package?

I'd prefer to keep same with other bean configuration classes, like `AadOAuth2ClientConfiguration`.
If we decide to change to `package-private / impl`, we'd better change them all.

Reply:
We should not add more public API. If this class won't be used by users directly we should put it in the impl package.